test/powershell/Modules/Microsoft.PowerShell.Utility/Wait-Event.Tests.ps1

```diff
@@ -1,17 +1,29 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 Describe "Wait-Event" -Tags "CI" {
-
     Context "Validate Wait-Event is waiting for events" {
-	It "Should time out when it does not receive a FakeEvent" {
-	    # Don't depend on Measure-Command
-$stopwatch = [System.Diagnostics.Stopwatch]::startNew()
-# Testing the timeout, so wait for an event that will never be
-# raised because it is fake
-Wait-Event -Timeout 1 -SourceIdentifier "FakeEvent"
-$stopwatch.Stop()
-$stopwatch.ElapsedMilliseconds | Should -BeGreaterThan 500
-$stopwatch.ElapsedMilliseconds | Should -BeLessThan 1500
-}
-}
+		It "Should time out when it does not receive a FakeEvent" {
+			# Don't depend on Measure-Command
+			$stopwatch = [System.Diagnostics.Stopwatch]::startNew()
+			# Testing the timeout, so wait for an event that will never be
+			# raised because it is fake
+			Wait-Event -Timeout 1 -SourceIdentifier "FakeEvent"
+			$stopwatch.Stop()
+			$stopwatch.ElapsedMilliseconds | Should -BeGreaterThan 500
+			$stopwatch.ElapsedMilliseconds | Should -BeLessThan 1500
+		}
+
+		It "Should be able to wait for an arbitrary -TimeSpan" {
+			# Don't depend on Measure-Command
+			$stopwatch = [System.Diagnostics.Stopwatch]::startNew()
+			# Testing the timeout, so wait for an event that will never be
+			# raised because it is fake
+			Wait-Event -TimeSpan ([Timespan]'00:00:00.25') -SourceIdentifier "FakeEvent"
+			$stopwatch.Stop()
+			$stopwatch.ElapsedMilliseconds | Should -BeGreaterThan 200
+			$stopwatch.ElapsedMilliseconds | Should -BeLessThan 300
+		}
+	}
```

Comment:
we should probably have an additional test for the transform. Replicating the test at line 16 with a _string_ value of "00:00:00.25" should do it. I don't think we need a test for a non-convertable value unless you want to validate that  you're getting a parameter bind/transform fail error

Reply:
@JamesWTruher a string value would be good.

It's a scenario where we'd need to use a -TransformScript, not just a property name (we'd check to see if it could be a timespan, then pull out the total seconds).

What is the preferred mechanism for using a ScriptBlock as an attribute value within the PowerShell codebase?

Also, Joel Brings up a fair point:

https://github.com/PowerShell/PowerShell/issues/20112#issuecomment-1676612421

Additionally, I feel it would be _very_ nice if we could make $This in the Transform be the command the ScriptBlock is tied to (but, alas, don't quite know how we'd be able to know that within an attribute).  It may also be _somewhat_ nice to have a string property used for metadata about the transform (e.g. a description).

Any ideas on how to make these happen?

Reply:
@SeeminglyScience had suggested these be two separate PRs (one for the transform attribute, one for Wait-Event's changes).

As eager as I am to see Wait-Event work with subsecond timeouts (it limits functionality of obs-powershell+ScriptDeck), please let me know if you'd prefer these be separate PRs.

Reply:
@StartAutomating I would agree with @SeeminglyScience that these should be two separate PRs starting with the ArgumentTransform (thanks for taking the feedback and creating it!).  It should probably have its own set of tests independent of this cmdlet.  Once that one is merged, this PR should be simplified to just use that attribute.